src/Umbraco.Web/umbraco.presentation/umbraco/Trees/XmlTree.cs

```diff
@@ -48,7 +47,7 @@ public XmlTree(SerializedTreeType treeType)
 
 		private void Init()
 		{
-            m_JSSerializer = new JSONSerializer { MaxJsonLength = int.MaxValue };
+            m_JSSerializer = new JavaScriptSerializer { MaxJsonLength = int.MaxValue };

```

Comment:
Instead of doing this - for now just throw: `NotSupportedException("Remove this code, this should never be used")` and eventually we'll get around to removing this entirely. I'm 99.9% sure that this code should never be executed.

You can basically do this everywhere you've changed the JSONSerializer  to be JavaScriptSerializer  in your PR. 

If you push these changes to this branch this PR will be updated to that code (don't submit a new PR for this )